src/Native/Unix/System.Native/pal_io.c

```diff
@@ -1186,37 +1205,67 @@ static int32_t CopyFile_ReadWrite(int inFd, int outFd)
     free(buffer);
     return 0;
 }
-#endif // !HAVE_FCOPYFILE
 
-int32_t SystemNative_CopyFile(intptr_t sourceFd, intptr_t destinationFd)
+static int32_t CopyFileMetadata(int inFd, int outFd, struct stat_ *sourceStat)
 {
-    int inFd = ToFileDescriptor(sourceFd);
-    int outFd = ToFileDescriptor(destinationFd);
-
-#if HAVE_FCOPYFILE
-    // If fcopyfile is available (OS X), try to use it, as the whole copy
-    // can be performed in the kernel, without lots of unnecessary copying.
-    // Copy data and metadata.
-    return fcopyfile(inFd, outFd, NULL, COPYFILE_ALL) == 0 ? 0 : -1;
-#else
-    // Get the stats on the source file.
+    // Now that the data from the file has been copied, copy over metadata
```

Comment:
This comment no longer makes sense now that it's been separated into its own function.  Let's delete the "Now that the data from the file has been copied, " part.